src/blocks/scratch3_sensing.js

```diff
@@ -40,4 +42,21 @@ Scratch3SensingBlocks.prototype.getMouseDown = function (args, util) {
     return util.ioQuery('mouse', 'getIsDown');
 };
 
+Scratch3SensingBlocks.prototype.current = function (args) {
+    var date = new Date();
+    switch (args.CURRENTMENU) {
+    case 'year': return date.getFullYear();
+    case 'month': return date.getMonth() + 1; // getMonth is zero-based
+    case 'date': return date.getDate();
+    case 'dayofweek': return date.getDay();
```

Comment:
Scratch 2.0 uses `day + 1`: Sunday is 1. Testing your implementation, I get 5 for Friday, in Scratch I get 6.